silk/tests/openthread/ot_test_multi_hop_traffic.py

```diff
@@ -0,0 +1,206 @@
+        self.network_data.panid = self.routers[0].panid
+
+        self.sed_children[0].join(self.network_data, "sleepy-end-device")
+        self.sed_children[0].set_sleep_poll_interval(POLL_INTERVAL)
+
+        for index in range(1, NUM_ROUTERS):
+            self.routers[index].join(self.network_data, "router")
+            self.sed_children[index].join(self.network_data, "sleepy-end-device")
+            self.sed_children[index].set_sleep_poll_interval(POLL_INTERVAL)
+            self.wait_for_completion(self.device_list)
+
+        for index in range(NUM_FED_CHILDREN):
+            self.fed_children[index].join(self.network_data, "end-node")
+            self.wait_for_completion(self.device_list)
+
+        for index in range(1, NUM_ROUTERS):
+            self.assertTrue(self.routers[index].get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_ROUTER)
+
+    @testcase.test_method_decorator
+    def test02_verify_link(self):
+        # Wait till first router has either established a link or
+        # has a valid "next hop" towards all other routers.
+        r1_rloc = int(self.routers[0].getprop(wpan.WPAN_THREAD_RLOC16), 16)
+
+        def check_r1_router_table():
+            router_table = self.routers[0].wpanctl("get", "get " + wpan.WPAN_THREAD_ROUTER_TABLE, 2)
+            router_table = wpan_table_parser.parse_router_table_result(router_table)
+            self.assertEqual(len(router_table), NUM_ROUTERS)
+            for entry in router_table:
+                verify(entry.rloc16 == r1_rloc or entry.is_link_established() or entry.next_hop != INVALID_ROUTER_ID)
+
+        verify_within(check_r1_router_table, ROUTER_TABLE_WAIT_TIME)
+
+    @testcase.test_method_decorator
+    def test03_Transmit_Receive(self):
+        # Send from the first router in the chain to the last one.
+        # Send from the SED child of the last router to the SED child of the first
+        # router.
+        # Send from the FED child of the first router to the FED child of the last
+        # router.
+        class AddressType(enum.Enum):
+            MLA = 0
+
+        addresses = [
+            (self.routers[0], self.routers[-1], AddressType.MLA, AddressType.MLA),
+            (self.sed_children[-1], self.sed_children[0], AddressType.MLA, AddressType.MLA),
+            (self.fed_children[0], self.fed_children[-1], AddressType.MLA, AddressType.MLA),
+        ]
+
+        timeout = 5
+        delay = 1
+        for i, (src, dst, src_address, dst_address) in enumerate(addresses):
+            port = random.randint(10000 + i * 100, 10099 + i * 100)
+            for msg_length in MSG_LENS:
+                message = random_string(msg_length)
+                src_address = src.ip6_mla
+                dst_address = dst.ip6_mla
+
+                dst.receive_udp_data(port, message, timeout)
+                time.sleep(delay)
+                src.send_udp_data(dst_address, port, message, src_address)
+
+                time.sleep(timeout - delay)
```

Comment:
Does it take longer to send larger message?

Reply:
No larger messages are also received in 1 minute